tests/core/cross/BUILD.bazel

```diff
@@ -24,6 +24,21 @@ go_binary(
     deps = [":platform_lib"],
 )
 
+go_binary(
+    name = "linux_cross_impure",
+    srcs = ["main.go"],
+    goarch = "amd64",
+    goos = "linux",
+    pure = "off",
```

Comment:
Just having second thoughts about semantics: If someone sets `pure = "off"`, is it okay for us to silently turn it on when no C toolchain is available? I'm wondering whether we should restrict this to unset `pure`. 

@illicitonion What do you think?

Reply:
It's a slightly weird corner - I _think_ any place that this would have errored before, but wouldn't error now, it's because `pure` doesn't actually need to be `off`, but I agree that ignoring an explicit instruction is probably bad.

I added this check, and a test showing we correctly fail.